src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/TypePreinit.cs

```diff
@@ -2095,6 +2135,8 @@ public override void WriteFieldData(ref ObjectDataBuilder builder, NodeFactory f
             {
                 builder.EmitPointerReloc(factory.SerializedFrozenObject(AllocationSite.OwningType, AllocationSite.InstructionCounter, this));
             }
+
+            public bool IsKnownImmutable => true;
```

Comment:
```suggestion
            public bool IsKnownImmutable => _methodPointed.Signature.IsStatic;
```
Per discussion, the immutability is transitive (the whole object graph needs to be immutable).